src/components/Profile.js

```diff
@@ -20,27 +20,13 @@ const useStyles = makeStyles((theme) => ({
 const Profile = () => {
 	const classes = useStyles()
 	const { currentUser } = useAuth()
-	const [currentUserName, setName] = useState('')
 	const email = currentUser.email
+	const {loading ,error,data} = useQuery(GET_PROFILE_NAME,{
+		variables:{email}
+	})
```

Comment:
You want to pull the api call outside of the component.
The data that needs to rendered is dependent on a successful response